### DISCUSSION TITLE
Is there a way to expose yazi's globals to `lua-language-server`?

### DISCUSSION THREAD

**Discussion by bvr-yr** [sxyazi/yazi#934] [1. Q&A]
Would be very handy since plugin system is :fire:

**Comment by sxyazi** [comment]
Unfortunately, there hasn't been much progress on this lately. Most methods and types are defined on the Rust side, and currently, there doesn't seem to be a good way to generate type definitions from them. 

I don't have enough time to maintain another set of documents besides the existing documentation, as it would double my workload, leaving me constantly struggling between Rust implementation, documentation, and Lua definitions. 

Perhaps we can revisit the possibility of finding a better way to maintain them once the plugin system stabilizes. For now, the plugin system is still in beta, meaning the API isn't fully finalized. 

I'm going to close this since it duplicates https://github.com/sxyazi/yazi/discussions/981
